core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/support/IncludeExclude.java

```diff
@@ -61,15 +65,35 @@
     private static final ParseField INCLUDE_FIELD = new ParseField("include");
     private static final ParseField EXCLUDE_FIELD = new ParseField("exclude");
     private static final ParseField PATTERN_FIELD = new ParseField("pattern");
+    private static final ParseField PARTITION_FIELD = new ParseField("partition");
+    private static final ParseField OF_FIELD = new ParseField("of");
 
     // The includeValue and excludeValue ByteRefs which are the result of the parsing
     // process are converted into a LongFilter when used on numeric fields
     // in the index.
-    public static class LongFilter {
+    public abstract static class LongFilter {
+        public abstract boolean accept(long value);
+   
+    }
+    
+    public class PartitionedLongFilter extends LongFilter {
+        private final ByteBuffer buffer = ByteBuffer.allocate(Long.BYTES);    
+
+        @Override
+        public boolean accept(long value) {
+            // hash the value to keep even distributions
+            buffer.putLong(0, value);
+            final int hashCode = StringHelper.murmurhash3_x86_32(buffer.array(), 0, Long.BYTES, StringHelper.GOOD_FAST_HASH_SEED);            
```

Comment:
you could call `BitMixer.mix64(value)`, which should have the same quality but might be a bit faster